pkg/ovs/openflow/ofctrl_bridge.go

```diff
@@ -216,21 +216,13 @@ func (b *OFBridge) CreateGroup(id GroupIDType) Group {
 }
 
 func (b *OFBridge) createGroupWithType(id GroupIDType, groupType ofctrl.GroupType) Group {
-	ofctrlGroup, err := b.ofSwitch.NewGroup(uint32(id), groupType)
-	if err != nil { // group already exists
-		ofctrlGroup = b.ofSwitch.GetGroup(uint32(id))
-	}
+	ofctrlGroup := &ofctrl.Group{ID: uint32(id), GroupType: groupType, Switch: b.ofSwitch}
```

Comment:
I think initiating the instance bypassing the provided constructor still leaves risk for future use. It would just look like the code is redundant with the construtor and other contributors may "optimize" it by introducing the bug back.

The real issue here is the bad implementation in `OFSwitch`, it provides concurency access for some data while not for other. I think we should fix that implementation for now instead, and revisit whether it's necessary to keep the caches in `OFSwitch` via another proposal/PR. If we choose to not keep that cache and makes `CreateGroup` a pure constructor, we should rename the name in this file to `NewGroup`.

Note that not only Group, but also Meter and Table have the same issue.